pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -17,13 +18,27 @@ import (
 )
 
 func testStableSystemOperatorStateTransitions(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
-	return testOperatorStateTransitions(events, []configv1.ClusterStatusConditionType{configv1.OperatorAvailable, configv1.OperatorDegraded})
+	// map from allowed interval regexps to exceptions
+	exceptions := map[*regexp.Regexp]string{
+		regexp.MustCompile(".*"): "We are not worried about Available=False or Degraded=True blips for stable-system tests yet.",
+	}
+
+	return testOperatorStateTransitions(events, []configv1.ClusterStatusConditionType{configv1.OperatorAvailable, configv1.OperatorDegraded}, exceptions)
 }
 
 func testUpgradeOperatorStateTransitions(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
-	return testOperatorStateTransitions(events, []configv1.ClusterStatusConditionType{configv1.OperatorAvailable, configv1.OperatorDegraded})
+	// map from allowed interval regexps to exceptions
+	exceptions := map[*regexp.Regexp]string{
+		regexp.MustCompile(".*condition/Degraded.*"): "We are not worried about Degraded=True blips for update tests yet.",
+
+		regexp.MustCompile(".*clusteroperator/authentication condition/Available reason/WellKnown_NotReady status/False[ :].*"):                                  "https://issues.redhat.com/browse/OCPBUGS-20056",
+		regexp.MustCompile(".*clusteroperator/control-plane-machine-set condition/Available reason/UnavailableReplicas status/False[ :].*"):                      "https://issues.redhat.com/browse/OCPBUGS-20061",
+		regexp.MustCompile(".*clusteroperator/kube-storage-version-migrator condition/Available reason/KubeStorageVersionMigrator_Deploying status/False[ :].*"): "https://issues.redhat.com/browse/OCPBUGS-20062",
+	}
```

Comment:
We are moving to structured intervals and this kind of regex matching has gotten us into trouble in other places in that effort. The intervals backing this test have not yet been ported but I can make that my next priority, I'll post here as soon as I can get that in. You'll need to do some kind of func based matching perhaps that can examine the type/fields of an interval for a match.

Big +1 in pinning those we know have problems. The [test results are pretty striking](https://sippy.dptools.openshift.org/sippy-ng/tests/4.15?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522never-stable%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522aggregated%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522should%2520not%2520change%2520condition%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&sort=desc&sortField=current_flake_percentage&view=Flakes).

Reply:
https://issues.redhat.com/browse/TRT-1314 I'll try to get to this this week.

Reply:
Looks like these were actually already ported, just not at the point you last ran tests in here, or at least not in the job I looked at.

A new interval for this would look like:

```json
        {
            "level": "Error",
            "locator": "clusteroperator/control-plane-machine-set",
            "message": "condition/Available reason/UnavailableReplicas status/False Missing 1 available replica(s)",
            "tempSource": "ClusterOperatorMonitor",
            "tempStructuredLocator": {
                "type": "ClusterOperator",
                "keys": {
                    "clusteroperator": "control-plane-machine-set"
                }
            },
            "tempStructuredMessage": {
                "reason": "UnavailableReplicas",
                "cause": "",
                "humanMessage": "Missing 1 available replica(s)",
                "annotations": {
                    "condition": "Available",
                    "reason": "UnavailableReplicas",
                    "status": "False"
                }
            },
            "from": "2023-10-18T19:40:38Z",
            "to": "2023-10-18T19:40:38Z"
        },
```

You'll want to match on the StructuredLocator.Type, StructuredLocator.Keys["clusteroperator"], StructuredMessageAnnotations condition+status+reason. I know reason is duplicated in there and I'm not sure which we keep, but we'll figure that out soon, I suspect it'll be the annotation not the actual field. Locator Keys and message Annotations should all have constants in monitorapi/types.go.